### PR TITLE
react reduxを使ってtodo appを作成しました。

### PR DESCRIPTION
**Pull request by Cody1009**
react reduxを使ってtodo appを作成しました。fetchやstate操作の部分のロジックはあまり変えていません。
あとrouterの問題で、完了済みのタスクを移すページに遷移はできるのですが、リロードを挟まないと完了済みのタスクだけ表示できるようにできません。
ご確認よろしくおねがいします。

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,227 +1,54 @@
...
                 </div>
             </BrowserRouter>
         )
     }
-
 }
+const mapStateToProps = state =>{
+    return{
+        todos: state.todos,
+        isFetching: state.isFetching
```
Comment (kawamurakazushi) [inline src/App.js]: Are you using `isFetching` ?

File: src/App.js [hunk 2]
```diff
@@ -1,227 +1,54 @@
...
                     />
                 </div>
             </BrowserRouter>
         )
     }
-
 }
+const mapStateToProps = state =>{
+    return{
+        todos: state.todos,
```
Comment (kawamurakazushi) [inline src/App.js]: is there a reason why you get the todos in `App` instead of `FormWrapper`?

File: src/components/Forms/TodosForm/TodosForm.js [hunk 3]
```diff
@@ -25,16 +25,21 @@ const styles = theme => ({
 
 });
 
-const completedTodos = (props) => {
+const TodosForm = (props) => {
     const {classes} = props;
-    const completed_todos_list = props.completedTodos.map(todo => {
+    const todos_list = props.todos.map(todo => {
```
Comment (kawamurakazushi) [inline src/components/Forms/TodosForm/TodosForm.js:30]: Small details but camelCase.

File: src/containers/FormWrapper/FormWrapper.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            />
+
+            {/* #############      FOOTER     ############ */}
+            <Route path="/" component={FooterForm}/>
+
+        </div>
+    );
+};
+
+
```
Comment (kawamurakazushi) [inline src/containers/FormWrapper/FormWrapper.js]: There is also a technique like this:

```js
const mapStateToProps = state => {
  return {
     todos: state.todos,
     completedTodos: state.todos.filter(...),
     notCompletedTodos: state.todos.filter(...),
  }
}
```

File: src/store/reducers/manipulateTodoState.js [hunk 5]
```diff
@@ -0,0 +1,155 @@
+import * as actionTypes from '../actions/actionTypes';
```
Comment (kawamurakazushi) [inline src/store/reducers/manipulateTodoState.js]: I think the file name could be `store/reducers/todos`.

File: src/App.js (hunk 1 above)
Comment (Cody1009) [reply src/App.js]: Ahh, actually not. I looked into some articles and sample code, and it used a state "isLoading" when dispatching async code. I thought it was conventional but it is not?

File: src/App.js (hunk 2 above)
Comment (Cody1009) [reply src/App.js]: I wanted to get todos and set it in  component did mount. I may change formWrapper into class component and use it there but what do you think?

File: src/components/Forms/TodosForm/TodosForm.js (hunk 3 above)
Comment (Cody1009) [reply src/components/Forms/TodosForm/TodosForm.js:30]: oops! Thank you!

File: src/containers/FormWrapper/FormWrapper.js (hunk 4 above)
Comment (Cody1009) [reply src/containers/FormWrapper/FormWrapper.js]: I did not know that but it looks clearer ! Thank you!

File: src/store/reducers/manipulateTodoState.js (hunk 5 above)
Comment (Cody1009) [reply src/store/reducers/manipulateTodoState.js]: I will change it !

File: src/App.js (hunk 1 above)
Comment (kawamurakazushi) [reply src/App.js]: It is conventional but if you not using it, you should delete it.

File: src/App.js (hunk 2 above)
Comment (kawamurakazushi) [reply src/App.js]: I think dolng that inside the FormWrapper is a little bit clearer to me rather than doing inside App.

File: src/App.js (hunk 1 above)
Comment (Cody1009) [reply src/App.js]: Ok, then I will delete it!

File: src/App.js (hunk 2 above)
Comment (Cody1009) [reply src/App.js]: I understood, I will change FormWrapper into class component and use componentDidMount there

**Pull request merged**